### PR TITLE
Improve docs, comments, and correctness

### PR DESCRIPTION
**Pull request by glenn-jocher**
## Summary

Full-repo correctness sweep fixing factual errors in docs, docstrings, comments, examples, and config, plus four narrowly-scoped latent code bugs. Every change is grounded in the current implementation, dataset YAMLs, manifests, or the Platform source of truth. 78 files, +186/−193 (net deletion).

### Docs/code mismatches
- Benchmark classification metric is `accuracy_top1`, not `accuracy_top5` (`TASK2METRIC`) — `modes/benchmark.md`, `modes/index.md`, `usage/python.md`
- Removed nonexistent `gpus:` config key; multi-GPU uses `device=[0,1,2,3]` — `guides/yolo-common-issues.md`
- `mode=segment` is invalid (segment is a task); fixed CLI example to `task=segment mode=predict`
- `batch_size` → `batch` (the actual arg name) in `guides/model-training-tips.md`
- `project`/`name` defaults are `None` (mode-based incrementing `train`, `train-2`), not `'runs'`/`'exp'` — `usage/cfg.md`
- TensorRT int8 end-to-end auto-disable is specific to TensorRT 10.3.0 on JetPack 6 (`exporter.py`), not `≤10.3.0`/`>10.3.0` — `guides/end2end-detection.md`
- Removed false claim that TensorRT INT8 auto-sets `dynamic=True` — `docs/macros/export-args.md`
- `Results.plot()` operates on/returns BGR (not RGB); `txt_color` is BGR — `modes/predict.md`
- Solutions API: `draw_specific_kpts()` returns `np.ndarray` (not `draw_specific_points()`/`None`), `plot_workout_information()` returns `int`, AIGym configures via `up_angle`/`down_angle`/`kpts` (no `pushup`/`pullup`/`abworkout` pose types), ObjectCropper does not support `show=True`, solutions help is `yolo solutions help`
- `plot_results()` takes a results CSV path, not a Results object — `integrations/jupyterlab.md`
- Ray Tune `close_mosaic` space is `tune.randint(0, 11)` (int bounds) — `integrations/ray-tune.md`
- Docker quickstart says six images (list has six) — `quickstart.md`

### Dataset docs vs YAMLs
- carparts-seg, crack-seg, package-seg: test/validation counts were swapped (and carparts train 3156 → 3516)
- dog-pose: split is train/val (not train/test), total ≈ 8,500 images
- COCO128 is the first 128 images of train2017 (not train+val)
- COCO-Pose eval server competition id 7403 (consistent with coco-pose.yaml and the same page)
- COCO12-Formats covers 12 of the supported formats (IMG_FORMATS has 14); fixed the equality assert that could never pass
- Caltech-101 video caption said Caltech-256

### Translation sync (README.zh-CN.md)
- Detection and OBB tables were missing the e2e metric columns; Segmentation/Pose headers were missing the `(e2e)` suffix; values now mirror README.md exactly
- Removed stray Reddit tracking parameter

### Broken/misleading links
- CodeQL workflow links pointed to nonexistent `codeql.yaml`; now use the actual `github-code-scanning/codeql` path (matches the badge on the same page)
- YOLOv5 GCP/AWS quickstart links pointed at unrelated integration pages; sparsify notice link text said sparseml; yolov5-seg link text/href mismatch; SolutionResults anchor pointed at SolutionAnnotator

### Docstring examples (verified at runtime)
- Corrected wrong example outputs in SAM modules: PatchEmbed `(1,14,14,768)`, PatchMerging `(4,784,128)`, ConvLayer `(1,64,56,56)`, BasicLayer `(1,3136,192)`, MemoryEncoder pos `(1,64,64,64)`, window_unpartition `(8,15,14,64)`, MBConv channel mismatch
- `_handle_deprecation` example output (`show_labels=False` for string `"False"`), `handle_yolo_settings` example used a nonexistent settings key

### Code fixes (latent bugs, narrowly scoped)
- `tiny_encoder.py`: `torch.utils.checkpoint(...)` called a module — now `torch.utils.checkpoint.checkpoint(...)` (TypeError whenever `use_checkpoint=True`)
- `sam3/encoder.py`: `x.dim == 4` compared a bound method — now `x.dim()`
- `callbacks/dvc.py`: OBB confusion matrix is `(nc+1)²` but background label was only appended for detect → IndexError; now `{"detect", "obb"}`
- `.github/workflows/cla.yml`: allowlist had a stray `[` and duplicated `pre-commit*` entry (typo present since introduction)

### Typos and stale comments
`funetuning`, `exponential standoff`, `addition optimizers`, `LabeI Studio`, `DPS`→`FPS` (3×), `pycotools`, stale `1E12 frames` note, stale shape/rows-cols comments, `validation model`→`mode`, copy-pasted nc-hardcoded NOTEs in YOLOE linear probing, stale PoseValidator Methods docstring, notebook comments saying COCO128 while training on COCO8

## Validation

- `git diff --check` clean; all changed Python compiles (`py_compile`)
- `ruff check` (repo flags) and `ruff format --check --line-length 120`: all 30 changed Python files pass
- Every changed docstring example literal re-derived at runtime; checkpoint/assert/dvc fixes exercised directly
- `actionlint .github/workflows/cla.yml` passes; `examples/tutorial.ipynb` parses as valid JSON
- `python docs/build_reference.py`: no drift in generated reference docs; `mkdocs build` passes
- README zh/en tables: column counts and all numeric values verified to match programmatically
- `yolo predict model=yolo26n.pt imgsz=320` smoke test and `pytest tests/test_python.py::test_predict_img` (7 passed)
- Edited external links checked (HTTP 200): CodeQL workflow, CodaLab competition, sparsify notice, yolov5-seg repo

## Verification sources

- `ultralytics/cfg/__init__.py`, `default.yaml`, `engine/exporter.py`, `engine/results.py`, `utils/tuner.py`, `utils/benchmarks.py`, solutions sources, dataset YAMLs
- Platform docs verified against the Portal app source (datasets route limit; training-time claim consistent with cloud-training.md cost examples and the Portal cost model)
- GitHub state via API: workflow inventory confirms no `codeql.yaml` (CodeQL runs as `github-code-scanning/codeql`)

## Review process

Candidate fixes were drafted from a full line-by-line audit of every tracked file, then each finding was independently re-verified against source before patching (83 accepted as-is, 25 revised, 7 rejected, 3 deferred). The final diff passed four independent review passes (factual re-verification, internal consistency/CI-readiness, and two adversarial code-review passes) with all actionable findings addressed.

## Intentionally not changed (needs maintainer/source-side decision)

- `platform/api/index.md` Download rate-limit row: Platform has no `/api/models/{id}/download` route (only `/files`), but the Platform rate limiter still maps the 30/min bucket to the old `/download` pattern — docs can't be made truthful without a Platform-side fix first
- `TT100K.yaml`: `names` contains 46 duplicated strings across 92 ids, so the documented `class_to_idx` inversion collapses to 175 classes; needs the authoritative upstream category list
- `Dockerfile-runner-cpu` registers the runner at org level while `Dockerfile-runner` uses repo level — may be intentional
- MobileSAM 632M-vs-611M encoder param inconsistency (both figures appear in the paper)

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
🧹 This PR is mainly a documentation and quality-of-life cleanup, improving accuracy across Ultralytics docs, examples, and comments, with a small code update to improve TinyViT/SAM checkpoint compatibility and developer clarity.

### 📊 Key Changes
- 📚 Updated many docs, READMEs, and examples to fix inaccuracies, outdated commands, typos, and broken or incorrect links.
- 📈 Refreshed YOLO26 benchmark tables in the Chinese README to include clearer end-to-end (`e2e`) metrics for detection, segmentation, pose, and OBB tasks.
- 🗂️ Corrected dataset documentation details, including split names/counts, dataset descriptions, and API parameter docs for datasets like MNIST160, COCO128, Dog-pose, Carparts, Crack, and Package segmentation.
- 🛠️ Improved usage guidance in multiple guides by fixing command arguments and examples, such as:
  - multi-GPU training now documented with `device=[...]`
  - benchmark classification metric corrected to `accuracy_top1`
  - plotting examples updated to use `results.csv`
  - CLI solution/help examples clarified
- 🤖 Clarified export and deployment notes, especially around TensorRT end-to-end behavior on JetPack 6 and related export settings.
- 🧪 Added or improved docstrings and inline comments across Python code for better maintainability and more accurate developer references.
- ⚙️ Included a small functional code improvement in SAM/TinyViT:
  - added safer checkpoint kwargs handling for PyTorch version compatibility
  - refined some train/inference behavior in TinyViT attention handling
- 🔍 Minor workflow/config cleanup:
  - simplified CLA bot allowlist
  - fixed Docker example comments
  - updated spelling ignore list
  - improved typing hints such as `str | None`

### 🎯 Purpose & Impact
- ✅ Makes the documentation more trustworthy and easier to follow for both new and experienced users.
- 🚀 Reduces confusion from outdated commands, wrong metric names, incorrect dataset stats, and broken links.
- 🧠 Helps developers by improving code comments, docstrings, and type hints, making the codebase easier to understand and maintain.
- 🔧 Improves compatibility and stability for some SAM/TinyViT workflows, especially around PyTorch checkpointing behavior.
- 🌍 Improves the overall user experience across docs, examples, CI references, platform docs, and multilingual README content.
- 📦 Low risk overall: most changes are documentation and comment fixes, with only a small number of lightweight code-level adjustments.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help streamline review; an engineer will assist shortly. Please review the checklist below to help ensure smooth integration ✨

-✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
-✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
-✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
-✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
-✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
-✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
-✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don’t hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: examples/YOLO-Series-ONNXRuntime-Rust/README.md [hunk 1]
```diff
@@ -36,9 +36,9 @@ This repository provides a [Rust](https://rust-lang.org/) demo showcasing key [U
        ```
 
 - **Option 2: Automatic Download**
-  - Use the `--features auto` flag with Cargo to let the build script handle downloading the library automatically:
+  - The `auto` feature of the `usls` dependency (already enabled in `Cargo.toml`) lets the build script handle downloading the library automatically:
     ```bash
-    cargo run -r --example yolo --features auto
+    cargo run -r -- --task detect --ver v8
```
Comment (UltralyticsAssistant) [inline examples/YOLO-Series-ONNXRuntime-Rust/README.md:41]: ⚠️ **HIGH**: This drops `--example yolo`, so `cargo run` will execute the package's default binary (or fail if the crate has no default bin) instead of the example this section is documenting. Keep the example selector and pass the runtime args after `--`.

**Suggested change:**
```suggestion
    cargo run -r --example yolo -- --task detect --ver v8
```

File: docs/en/help/CI.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ Here's a brief description of our CI actions:
 - **[CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml):** This is our primary CI test that involves running unit tests, linting checks, and sometimes more comprehensive tests depending on the repository.
 - **[Docker Deployment](https://github.com/ultralytics/ultralytics/actions/workflows/docker.yml):** This test checks the deployment of the project using Docker to ensure the Dockerfile and related scripts are working correctly.
 - **[Broken Links](https://github.com/ultralytics/ultralytics/actions/workflows/links.yml):** This test scans the codebase for any broken or dead links in our markdown or HTML files.
-- **[CodeQL](https://github.com/ultralytics/ultralytics/actions/workflows/codeql.yaml):** CodeQL is a tool from GitHub that performs semantic analysis on our code, helping to find potential security vulnerabilities and maintain high-quality code.
+- **[CodeQL](https://github.com/ultralytics/ultralytics/actions/workflows/github-code-scanning/codeql):** CodeQL is a tool from GitHub that performs semantic analysis on our code, helping to find potential security vulnerabilities and maintain high-quality code.
```
Comment (UltralyticsAssistant) [inline docs/en/help/CI.md:18]: 💡 **MEDIUM**: This new target does not look like a valid GitHub Actions or Code Scanning URL. `actions/workflows/github-code-scanning/codeql` is not the normal route for either a workflow file or the security dashboard, so the updated CodeQL links in this page will likely 404 for readers.

File: ultralytics/models/sam/modules/utils.py [hunk 3]
```diff
@@ -276,7 +276,7 @@ def window_unpartition(windows: torch.Tensor, window_size: int, pad_hw: tuple[in
         >>> hw = (15, 14)  # Original height and width
         >>> x = window_unpartition(windows, window_size=8, pad_hw=pad_hw, hw=hw)
         >>> print(x.shape)
-        torch.Size([1, 15, 14, 64])
+        torch.Size([8, 15, 14, 64])
```
Comment (UltralyticsAssistant) [inline ultralytics/models/sam/modules/utils.py:279]: 📝 **LOW**: The updated example shape is inconsistent with `window_unpartition()`'s batching logic for the shown `window_size`, `pad_hw`, and `hw` values. Recovering from 8x8 windows on a padded 16x16 image should collapse windows back to the original batch size, not expand it to 8.

**Suggested change:**
```suggestion
        torch.Size([1, 15, 14, 64])
```

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Mostly a clean documentation/comment pass, but I found a couple of user-facing regressions: one likely broken CodeQL link, one Rust example command that will run the wrong target, and one incorrect SAM doc example output shape.

💬 Posted 3 inline comments

File: examples/YOLO-Series-ONNXRuntime-Rust/README.md (hunk 1 above)
Comment (glenn-jocher) [reply examples/YOLO-Series-ONNXRuntime-Rust/README.md:41]: The crate has no `examples/` directory and no `[[example]]` targets in Cargo.toml — only `src/main.rs` — so `cargo run -r --example yolo` fails with "no example target named yolo". The default binary is the demo itself, and the Usage section below (line 60+) already invokes it the same way: `cargo run -r -- --task detect --ver v8`. The old command was the broken one.

File: docs/en/help/CI.md (hunk 2 above)
Comment (glenn-jocher) [reply docs/en/help/CI.md:18]: Verified against live GitHub state: `gh api repos/ultralytics/ultralytics/actions/workflows` lists CodeQL at path `dynamic/github-code-scanning/codeql` (default-setup code scanning), and https://github.com/ultralytics/ultralytics/actions/workflows/github-code-scanning/codeql returns 200 and renders the CodeQL runs. The previous `codeql.yaml` target 404s at the API level — no such workflow file exists in .github/workflows/. The badge at line 29 of this same page already uses the `github-code-scanning/codeql` path.

File: ultralytics/models/sam/modules/utils.py (hunk 3 above)
Comment (glenn-jocher) [reply ultralytics/models/sam/modules/utils.py:279]: Executed at runtime: `window_unpartition(torch.rand(32, 8, 8, 64), 8, (16, 16), (15, 14)).shape` returns `torch.Size([8, 15, 14, 64])`. With window_size=8 on a padded 16×16 image, each image produces (16/8)×(16/8)=4 windows, so 32 windows correspond to B=32/4=8 images — the function recovers batch size 8, not 1. The old `[1, 15, 14, 64]` literal was the bug this change fixes.

**Comment by UltralyticsAssistant** [comment]
Merged! 🎉 Thanks @glenn-jocher for this thoughtful cleanup and polish across the codebase and docs.

“Details make perfection, and perfection is not a detail.” — Leonardo da Vinci

This PR is a great example of that idea in practice: sharper docs, corrected examples and metrics, cleaner developer references, and a small but meaningful TinyViT/SAM compatibility improvement all add up to a smoother, more trustworthy experience for the entire Ultralytics community. Really appreciate the care here—these quality-of-life improvements make a big difference.
